# Дашборд для Дзена

## Описание

Автоматизация подсчета информации о работе пользователей с карточками различных статей в зависимости от темы и источника карточкек.

## Загрузка данных

In [1]:
# импортируем библиотеки
import pandas as pd
from sqlalchemy import create_engine, text

# настраиваем соединение с базой данных
db_config = {'user': 'praktikum_student', # имя пользователя
            'pwd': 'Sdf4$2;d-d30pp', # пароль
            'host': 'rc1b-wcoijxj3yxfsf3fs.mdb.yandexcloud.net',
            'port': 6432, # порт подключения
            'db': 'data-analyst-zen-project-db'} # название базы данных

connection_string = 'postgresql://{}:{}@{}:{}/{}'.format(db_config['user'],
                                                db_config['pwd'],
                                                db_config['host'],
                                                db_config['port'],
                                                db_config['db'])

# подключаемся к базе данных
engine = create_engine(connection_string)

In [2]:
query = 'SELECT * FROM dash_visits'

dash_visits = pd.io.sql.read_sql(query, con = engine)

In [3]:
dash_visits.head()

record_id item_topic source_topic age_segment                  dt  visits
0    1040597     Деньги         Авто       18-25 2019-09-24 18:32:00       3
1    1040598     Деньги         Авто       18-25 2019-09-24 18:35:00       1
2    1040599     Деньги         Авто       18-25 2019-09-24 18:54:00       4
3    1040600     Деньги         Авто       18-25 2019-09-24 18:55:00      17
4    1040601     Деньги         Авто       18-25 2019-09-24 18:56:00      27

## Предобработка данных

In [4]:
dash_visits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30745 entries, 0 to 30744
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   record_id     30745 non-null  int64         
 1   item_topic    30745 non-null  object        
 2   source_topic  30745 non-null  object        
 3   age_segment   30745 non-null  object        
 4   dt            30745 non-null  datetime64[ns]
 5   visits        30745 non-null  int64         
dtypes: datetime64[ns](1), int64(2), object(3)
memory usage: 1.4+ MB


С типами данных все в порядке.

In [5]:
dash_visits.describe()

record_id        visits
count  3.074500e+04  30745.000000
mean   1.055969e+06     10.089673
std    8.875461e+03     19.727601
min    1.040597e+06      1.000000
25%    1.048283e+06      1.000000
50%    1.055969e+06      3.000000
75%    1.063655e+06     10.000000
max    1.071341e+06    371.000000

In [6]:
dash_visits.duplicated().sum()

0

Явные дубликаты не найдены. 

## Посмотрим данные по столбцам

In [7]:
print(dash_visits['item_topic'].value_counts())
print('\n')
print(dash_visits['source_topic'].value_counts())
print('\n')
print(dash_visits['age_segment'].value_counts())
print('\n')
print(dash_visits['dt'].min(), '||', dash_visits['dt'].max())

Отношения             1536
Интересные факты      1535
Наука                 1505
Подборки              1456
Полезные советы       1424
Общество              1422
Россия                1385
История               1363
Семья                 1287
Путешествия           1247
Деньги                1234
Женщины               1230
Дети                  1229
Туризм                1206
Здоровье              1203
Красота               1193
Культура              1160
Юмор                  1129
Искусство             1119
Рассказы              1109
Психология            1056
Скандалы              1023
Знаменитости           976
Женская психология     914
Шоу                    804
Name: item_topic, dtype: int64


Семейные отношения    1822
Россия                1687
Знаменитости          1650
Полезные советы       1578
Путешествия           1563
Кино                  1505
Дети                  1459
История               1437
Семья                 1405
Одежда                1379
Здоровье              

## Выгрузка данных в CSV

In [8]:
dash_visits.to_csv('dash_visits.csv', index=False)

## Дашборд

<https://public.tableau.com/app/profile/elizaveta.taverova/viz/Book_for_yandex/Dashboard?publish=yes>

## Презентация

<https://disk.yandex.ee/i/9WU9gMMWXdGZ9g>

## Выводы

### Cколько взаимодействий пользователей с карточками происходит в системе с разбивкой по темам карточек?

Топ 5 популярных темам карточек:

- Наука 
- Отношения 
- Интересные факты 
- Общество
- Подборки

### Как много карточек генерируют источники с разными темами?

Всего 30745 карточек.

Топ 5 карточек по источникам:

- Семейные отношения
- Россия
- Полезные советы
- Путешествия
- Знаменитости

### Как соотносятся темы карточек и темы источников?

Топ 5 соотношений:

- Рассказы-Путешествия
- Общество-Россия
- Наука-Кино
- Подборки-Полезные советы
- Общество-Семейные отношения